### ISSUE TITLE
Incorrect `IDE0042` when tuple variable has usages

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#66994]
**Version Used**: 
Latest `main`

**Steps to Reproduce**:
```cs
(int A, int B) ints = (1, 1);
M(ints);

void M((int, int) i)
{

}
```

**Diagnostic Id**:
IDE0042

**Expected Behavior**:
`IDE0042` is not reported because `ints` variable has usages

**Actual Behavior**:
`IDE0042` is reported and fixing it produces the following code:
```cs
(int A, int B) = (1, 1);
M(ints);

void M((int, int) i)
{

}
```
Since `ints` is no longer defined a compiler error now appears.

**Comment by CyrusNajmabadi** [comment]
It's this just for top level code?

**Comment by DoctorKrolic** [comment]
Yes
![devenv_BrIWHwSDu1](https://user-images.githubusercontent.com/70431552/220709665-2d36976b-845b-43b6-8512-97d3bb2095c1.gif)
